### ISSUE TITLE
Filter incoming/outgoing payments by tx id

### ISSUE THREAD

**Issue by arnuschky** [monero-ecosystem/monero-python#53]
Is it possible to filter the incoming/outgoing payments by tx id? I am surprised that this works only by payment id - most of the time I only have the tx id.

**Comment by emesik** [comment]
Good idea. The RPC wallet [has even a method for it](https://www.getmonero.org/resources/developer-guides/wallet-rpc.html#get_transfer_by_txid).

**Comment by arnuschky** [comment]
Temp workaround:
```python
incoming = w.incoming()

def tx_id_in_incoming(tx_id):
    return len(list(filter(lambda i: i.transaction.hash == tx_id, incoming))) == 1
```

**Comment by emesik** [comment]
`>= 1` as you may receive multiple payments within one transaction.

**Comment by emesik** [comment]
Implemented in d44d4fd2ac1422527a7d3e64e09a65741280e57c. Needs docs and will be released in 0.6
